extensions/mssql/src/dashboard/bookWidget.ts

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as nls from 'vscode-nls';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import {
+	BookContributionProvider
+} from './bookExtensions';
+
+const localize = nls.loadMessageBundle();
+
+export function registerBooksWidget(bookContributionProvider: BookContributionProvider): void {
+	azdata.ui.registerModelViewProvider('books-widget', async (view) => {
+		const container = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'column',
+			width: '100%',
+			height: '100%',
+			alignItems: 'left'
+		}).component();
+		const bookslocationContainer = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'column',
+			width: '270px',
+			height: '100%',
+			alignItems: 'left',
+			position: 'absolute'
+		}).component();
+		const bookRow = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'row'
+		}).component();
+		const tsgbooklink = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: localize('troubleshootingBooks', 'Troubleshooting Book'),
```

Comment:
You should get the label and title for each of the book contributions, not hardcode one. Similarly we should plan ahead for when we have > 1 book by mapping from contributions to buttons:
```ts
let items = bookContributionProvider.contributions.map(contribution => {
    const tsgbooklink = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({
			label: localize('troubleshootingBooks', 'Troubleshooting Book'),
			title: localize('troubleshootingBooksTitle', 'Troubleshooting Book'),
		}).component();
		tsgbooklink.onDidClick(() => {
			PromptForFolder(bookContributionProvider);
		});
    return tsgBookLink;
});
bookRow.addItems(items,, {
			CSSStyles: {
				'width': '100%',
				'color': '#0078d4',
				'text-decoration': 'underline',
				'padding-top': '10px',
				'text-align': 'left'
			}
		});
```